_posts/2025-07-25-model-registry-integration.md

```diff
@@ -0,0 +1,445 @@
+- **Solution:** Aggressive Docker/`kind` cleanup (`sudo docker system prune -a --volumes -f`), ensuring `kind` is installed correctly (refer to `kind`'s [installation guide](https://kind.sigs.k8s.io/docs/user/quick-start/#installation)), and a full system logout/login to refresh the environment.
+
+- KFP Core Component Crashes (`metadata-grpc-deployment CrashLoopBackOff` with `Exit Code 139` - Segmentation Fault): This often indicates an incompatibility between the `ml_metadata_store_server` image and your host's kernel.
+
+- **Solution:** Patch the local KFP manifests to use an older, more compatible image version (e.g., `1.13.0` or `1.12.0`) in `~/kfp-local-manifests/manifests/kustomize/base/metadata/base/metadata-grpc-deployment.yaml.`
+
+- PersistentVolumeClaim (PVC) `Pending`: This means the cluster's local storage provisioner isn't working.
+
+- **Solution:** A full `kind` and Docker reset (`sudo systemctl stop docker, sudo systemctl start docker`) often resolves this by restarting the `kind`'s default local path provisioner.
+
+- "Pipeline version creation failed" / "Cannot find context" in KFP UI: These are symptoms of KFP's core components (especially ML Metadata and the KFP API server) not being fully `Running` and `READY`.
+
+- **Solution:** Patience and verifying `kubectl get pods -n kubeflow` are key.
+
+- Model Registry Pods Missing / `CreateContainerConfigError`: This usually indicates a configuration problem preventing the container from starting. While transient network issues can sometimes cause similar symptoms, for `CreateContainerConfigError`, a manifest fix is often needed. If persistent, ensure your Model Registry deployment manifests (e.g., from your cloned repository) are up-to-date, as an outdated fork might lead to incompatibilities with other Kubeflow components. We found success deploying the Model Registry using its Kustomize manifests from the `kubeflow/model-registry` repository's `overlays/db` directory.
+
+- `Model_registry.exceptions.StoreError: Version X already exists`: This error occurs when you try to register a model version with a `model_name` and `model_version_name `combination that already exists in the Model Registry.
+- **Solution:** This indicates the Model Registry is working as intended! Simply provide a new, unique `model_version_name` for your pipeline run (e.g., `v1.0.1`, `v2.0.0`, or `v1.0.0-run-timestamp`).
+
+
+# Deployment Steps:
+
+Install `kind` (if not already installed via dnf):
+
+```
+sudo dnf install kind # Recommended for Fedora
+# OR manual install if dnf fails (e.g., if 'No match for argument: kind'):
+# cd ~
+# curl -Lo kind https://kind.sigs.k8s.io/dl/v0.22.0/kind-linux-amd64
+# chmod +x kind
+# sudo mv kind /usr/local/bin/kind
+# Perform a full system logout and login after any kind installation/update.
+```
+
+Clone Kubeflow Pipelines Repository:
```

Comment:
A lot of this section seems to duplicate what the install script does. We should be able to just use the install script and also just use the kubeflow/manifests repo for installation.